File: backend/database/migrations/run_postes_migration.js
const mysql = require('mysql2/promise');
const fs = require('fs');
const path = require('path');

async function runMigration() {
  const connection = await mysql.createConnection({
    host: 'localhost',
    user: 'root',
    password: '',
    database: 'fs_hub'
  });

  try {
    console.log('Running migration: Create postes table');
    
    // Read migration file
    const migrationPath = path.join(__dirname, 'create_postes_table.sql');
    const migrationSQL = fs.readFileSync(migrationPath, 'utf8');
    
    // Execute migration
    await connection.execute(migrationSQL);
    console.log('✅ Postes table migration completed successfully!');
    
  } catch (error) {
    if (error.code === 'ER_TABLE_EXISTS_ERROR') {
      console.log('✅ Postes table already exists - no migration needed');
    } else {
      console.error('❌ Migration failed:', error);
    }
  } finally {
    await connection.end();
  }
}

runMigration();

File: backend/database/migrations/run_postes_migration.dart
import 'dart:io';

Future<void> main() async {
  try {
    print('Running migration: Create postes table');
    
    // Read SQL file
    final file = File('create_postes_table.sql');
    if (!await file.exists()) {
      print('❌ SQL file not found: create_postes_table.sql');
      return;
    }
    
    final sqlContent = await file.readAsString();
    print('SQL content loaded. Please run this SQL manually in your MySQL client:');
    print('\n${'='*50}');
    print(sqlContent);
    print('='*50);
    print('\nOr use MySQL command line:');
    print('mysql -u root -p fs_hub < create_postes_table.sql');
    
  } catch (e) {
    print('❌ Error: $e');
  }
}
